lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -181,29 +181,42 @@ def process_batch(api_endpoint, prospects, eager_submit)
     [submissions, errors]
   end
 
-  # Deletes a list of prospects from Pardot. For Pardot API documentation, see
-  # http://developer.pardot.com/kb/api-version-4/prospects/#using-prospects.
-  # @param [Array[Integer]] prospect_ids of the prospects to be deleted.
-  # @return [Array[Integer]] the set of prospect_ids that failed to be deleted.
-  def self.delete_prospects(prospect_ids)
-    failed_prospect_ids = []
-
-    return prospect_ids unless CDO.rack_env? :production
-    return prospect_ids if prospect_ids.length > MAX_PROSPECT_DELETION_BATCH_SIZE
-
-    prospect_ids.each do |prospect_id|
-      url = "#{PROSPECT_DELETION_URL}/#{prospect_id}"
-      post_request_with_auth url
-    rescue RuntimeError => e
-      if e.message =~ /Pardot request failed with HTTP/
-        failed_prospect_ids << prospect_id
-        next
-      else
-        raise e
-      end
+  # Deletes multiple prospects from Pardot using an email address.
+  def self.delete_prospects_by_email(email)
+    retrieve_pardot_ids_by_email(email).each do |id|
+      delete_prospect_by_id(id)
     end
+  end
+
+  # Deletes a prospect from Pardot using Pardot Id.
+  # This method only runs in the production environment to avoid accidentally deleting prospect.
+  # @param [Integer, String] pardot_id of the prospects to be deleted.
+  def self.delete_prospect_by_id(pardot_id)
+    if CDO.rack_env != :production
+      log "#{__method__} only runs in production. The current environment is #{CDO.rack_env}."
+      return
+    end
+
+    # @see http://developer.pardot.com/kb/api-version-4/prospects/#using-prospects
+    post_with_auth_retry "#{PROSPECT_DELETION_URL}/#{pardot_id}"
+  rescue StandardError => e
+    # If the input pardot_id does not exist, Pardot will response with
+    # HTTP code 400 and error code 3 "Invalid prospect ID" in the body.
+    return if e.message =~ /Pardot request failed with HTTP 400/
```

Comment:
Should we put this message in `pardot_helpers`?

Reply:
Yes. I put a TODO in `pardot_helpers` (also in this PR) to wrap both the HTTP status code and actual Pardot error in a custom exception. Will address that in another PR.
https://github.com/code-dot-org/code-dot-org/blob/93191ee66d9d288b05f4f85a779186d94ebb945d/lib/cdo/contact_rollups/v2/pardot_helpers.rb#L76-L79